src/runtime/GlobalObjectBuiltinPromise.cpp

```diff
@@ -158,19 +156,17 @@ static Value builtinPromiseAll(ExecutionState& state, Value thisValue, size_t ar
             // Set the [[Values]] internal slot of resolveElement to values.
             // Set the [[Capabilities]] internal slot of resolveElement to resultCapability.
             // Set the [[RemainingElements]] internal slot of resolveElement to remainingElementsCount.
-            Object* alreadyCalled = new Object(state);
-            alreadyCalled->defineOwnProperty(state, strings->value, ObjectPropertyDescriptor(Value(false), ObjectPropertyDescriptor::AllPresent));
+            bool* alreadyCalled = new (PointerFreeGC) bool(false);
```

Comment:
`EncodedValue` could hold primitive values like boolean at here.
IMHO how about using `EncodedValue` for primitive types instead of allocating it's value on heap?

Reply:
Because it should be shared by each {resolve, reject} function.
so that's why spec defines alreadyCalled as Record not primitive value

Reply:
Oh.. I see

Reply:
In jerryscript we used a bit in the Promise itself to represent the already called, and only allocate a similar boolean for thenable jobs (since any number of thenable jobs can be created and resolved for a single Promise).

Reply:
@zherczeg that idea is also good